### ISSUE TITLE
Duplicate requirement in Pipfile.lock when case differs 

### ISSUE THREAD

**Issue by jacebrowning** [kennethreitz/pipenv#229]
With 3.4.1, `$ pipenv install --lock mkdocs pdoc` produces a `Pipfile.lock` containing:

```
        "markdown": {
            "hash": "sha256:812ec5249f45edc31330b7fb06e52aaf6ab2d83aa27047df7cb6837ef2d269b6",
            "version": "==2.4.1"
        },
        "Markdown": {
            "hash": "sha256:0ac8a81e658167da95d063a9279c9c1b2699f37c7c4153256a458b3a43860e33",
            "version": "==2.6.8"
        },
```

which fails on reinstall:

```
$ pipenv install
No package provided, installing all dependencies.
Pipfile found at /private/tmp/Pipfile. Considering this to be the project home.
Installing dependencies from Pipfile.lock...
An error occured while installing!
Double requirement given: Markdown==2.6.8 (from -r /var/folders/ly/0k1p93rj1q56hk9rnk5j774h0000gn/T/tmpfc2fvv8a-requirements.txt (line 9)) (already in markdown==2.4.1 (from -r /var/folders/ly/0k1p93rj1q56hk9rnk5j774h0000gn/T/tmpfc2fvv8a-requirements.txt (line 8)), name='Markdown')
```

**Comment by nateprewitt** [comment]
Yep, this is definitely unintended @jacebrowning. The casing issue is easy to fix here, the thing that may need a bit more thought is versions. `mkdocs` requires Markdown>=2.3.1, where `pdoc` requires Markdown<=2.5.0. Fortunately 2.4.1 meets both of these requirements and works, but I'm concerned about a case that we choose the last downloaded file and end up with an incompatible dependency.

I'm going to have to read up on whether pip auto-detects this sort of things, and what our options are from third-part or pypa libraries. If anyone has background here and would like to share or propose a solution, that would be helpful.

**Comment by jacebrowning** [comment]
If pip doesn't detect this kind of behavior, maybe https://github.com/rbarrois/python-semanticversion (or another semver library) could help do something like this pseudocode:

```python
spec = Spec('>=2.3.1', '<=2.5.0')
for version in available_pip_downloads:
    if Version(version) in spec:
        return version
```

**Comment by nateprewitt** [comment]
This should be resolved with #242 and will be released in 3.5.0.
